src/common/snippets/include/snippets/pass/tokenization.hpp

```diff
@@ -70,94 +74,21 @@ class SnippetsTokenization : public ov::pass::ModelPass {
-            return m_data_ptr_gpr_count;
-        }
-
-        [[nodiscard]] bool get_split_m_dimension() const {
-            return m_split_m_dimension;
-        }
-
-        [[nodiscard]] bool get_mha_token_enable_transpose_on_output() const {
-            return m_mha_token_enable_transpose_on_output;
-        }
-
-        [[nodiscard]] bool is_dynamic_mha_token_enabled() const {
-            return m_is_dynamic_mha_token_enabled;
-        }
-
-        [[nodiscard]] std::set<size_t> get_mha_supported_transpose_ranks() const {
-            return m_mha_supported_transpose_ranks;
-        }
-
-        [[nodiscard]] const CanBeFusedAsPostOpPred& get_can_be_fused_as_postop() const {
-            return m_can_be_fused_as_postop;
-        }
-
-    private:
-        size_t m_concurrency = 0;
-        // The number of gpr that can be used as data pointers for data nodes (Parameter (and non-Scalar Constants),
-        // Result, Buffers with the same ID)
-        size_t m_data_ptr_gpr_count = 0;
-        // True if "SplitDimensionM" optimization is enabled. Otherwise, it's disabled.
-        bool m_split_m_dimension = true;
-        // False if Transpose on output isn't tokenized in MHA Tokenization.
-        // Otherwise, it may be fused into Subgraph if possible
-        // TODO [111813]: Remove please when the ticket 111813 is implemented
-        bool m_mha_token_enable_transpose_on_output = true;
-        // If True, MHA pattern with dynamic nodes will be tokenized
-        // Otherwise dynamic MHA won't be tokenized
-        // Currently, the flag can be set to `True` only for testing purposes.
-        bool m_is_dynamic_mha_token_enabled = true;
-        // Set of supported Transpose shape ranks for tokenization in MHATokenization pass.
-        // Note that in general Snippets support Transpose of any ranks.
-        // But at the moment Transpose is used only in MHA pattern where 3D and 4D tensors are supported.
-        std::set<size_t> m_mha_supported_transpose_ranks = {3, 4};
-        // Predicate that checks if the node can be fused as MatMul post-op.
-        // It is currently used only in TokenizeMLPSeqSnippets
-        CanBeFusedAsPostOpPred m_can_be_fused_as_postop = nullptr;
-    };
-
-    explicit SnippetsTokenization(Config config) : m_config(std::move(config)) {}
+    explicit SnippetsTokenization(TokenizationConfig config,
+                                  CommonOptimizations::Config common_config,
+                                  TokenizeMHASnippets::Config mha_config,
+                                  TokenizeMLPSeqSnippets::Config mlp_seq_config)
+        : m_config(config),
+          m_common_optimizationss_config(common_config),
+          m_mha_config(std::move(mha_config)),
+          m_mlp_seq_config(std::move(mlp_seq_config)) {}
     bool run_on_model(const std::shared_ptr<ov::Model>& m) override;
 
 private:
-    Config m_config;
+    TokenizationConfig m_config;
```

Comment:
Probably, `m_tokenization_config`. From the first glance without any context, it is not clear why it is named just `m_config`